### ISSUE TITLE
Option to disable/remove "show mnemonic seed" after wallet setup

### ISSUE THREAD

**Issue by infernix** [Samourai-Wallet/samourai-wallet-android#130]
I'd like to be able to remove the option to display mnemonic seed. I don't quite understand why the option exists to show it after the initial setup. It means that it's stored somewhere in plaintext, correct? Or is this not possible when using a passphrase?

If it really can't be removed due to key derivation, at least the option should exist to never show it. It can come with big disclaimers but I feel it shouldn't be displayable after the initial setup. There can be cases where someone can snap a picture of it unbeknownst to me, or against my will.

**Comment by SamouraiDev** [comment]
It is not stored anywhere in plaintext. It is decrypted and displayed upon demand. The passphrase, however, is not available on demand and must be remembered by the user.

**Comment by infernix** [comment]
OK, so then it's stored encrypted. But does it have to be? Can it be forgotten?

Because I don't like it that this is possible:

- Open wallet, do something
- Go to home screen
- Lock phone
- Unlock phone
- Use "task manager" to switch back to wallet
- Directly open settings > Show mnemonic

I'm not prompted for pin, passphrase or anything else. You could make the argument that I should lock the phone with a pin etc, but that doesn't make this any less of a "vulnerability".

**Comment by SamouraiDev** [comment]
The app does have a timeout feature which will disallow access after a period of inactivity via enforcement of PIN entry.

**Comment by SamouraiDev** [comment]
In the case of Samourai, which imposes the use of a BIP39 passphrase for the wallets it creates, the mnemonic alone cannot restore the wallet.

**Comment by infernix** [comment]
I understand that too. But that doesn't change my point?

**Comment by infernix** [comment]
To further drive home my point, if I restore a wallet without a passphrase, or if I use one without a passphrase for plausible deniability (and another one with a passphrase as the "real" one), all that's needed is one peek or screenshot or photo of it. So I'd really like to see an option to "forget" it (provided that it's not needed anywhere) or otherwise opt to permanently disable viewing it.

**Comment by LaurentMT** [comment]
Just a shower thought: may be another option would be to ask for the PIN before displaying the mnemonic.

**Comment by jonathancross** [comment]
I agree with @infernix - making the seed available (after the user has recorded it) reduces security of the wallet.  Such an option should be difficult to trigger.  The **PIN _and_ Passphrase** should be required before showing it.  Adding in a 1 minute delay and a huge warning would also be good.
